ScnSoft-Education/LeetCodeTraining/SortColors/SortColorsSolution.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace LeetCodeTraining.SortColors
+{
+    //https://leetcode.com/problems/sort-colors/
+    public static class SortColorsSolution
+    {
+        public static void SortColors(int[] nums)
+        {
+            int rptr = 0;
+            int bptr = nums.Length;
+
+            for (int i = 0; i < bptr; i++)
+            {
+                if (nums[i] == 0)
+                {
+                    Swap(rptr, i);
+                    rptr++;
+                    continue;
+                }
+                if (nums[i] == 2)
+                {
+                    Swap(i, bptr - 1);
+                    bptr--;
+                    if (nums[i] != 1)
```

Comment:
If is redundant, because nums[i] is 2 in this block of the code and always != 1...

Reply:
Hello @shomzascnsoft 
I believe it's not the case there, because previously the values were swapped and therefore nums[i] could become 1 as a result of this Swap.